app/uk/gov/hmrc/eoricommoncomponent/frontend/controllers/subscription/WhatIsYourEoriController.scala

```diff
@@ -98,11 +101,20 @@ class WhatIsYourEoriController @Inject() (
               whatIsYourEoriView(formWithErrors, isInReviewMode, UserLocation.isRow(requestSessionData), service)
             )
           ),
-        formData => submitNewDetails(formData, isInReviewMode, service)
+        formData => {
+          val eori = eoriWithCountry(formData.eoriNumber)
+          checkEoriNumberConnector.check(CheckEoriRequest(eori)).flatMap {
+            case Some(head :: _) if head.valid =>
+              submitEori(formData, isInReviewMode, service)
+            case Some(head :: _) =>
+              Future.successful(Redirect(routes.WhatIsYourEoriCheckFailedController.displayPage(head.eori, service)))
+            case _ => throw new MissingCheckResponseException
```

Comment:
Im not sure throwing an Exception is what we want here. Play will catch this and return a 500 to the user. Which is an InternalServiceError.  I think we really want to be returning a 400 BadRequest.

Reply:
This case should never happen (The connector will return None in the worse case after catching any exceptions). HMRC play apps catch all exceptions in a support page, with the message goes to the kibana logs.  When u r done pls call me on slack and we can discuss.

Reply:
I've tracked down the error handler that will catch this exception. It doesn't do much:  https://github.com/hmrc/eori-common-component-frontend/blob/main/app/uk/gov/hmrc/eoricommoncomponent/frontend/CdsErrorHandler.scala